collect_app/src/androidTest/java/org/odk/collect/android/preferences/qr/ConfigureWithQRCodeTest.java

```diff
@@ -168,19 +167,19 @@ public Bitmap generateQRBitMap(String data, int sideLength) throws IOException,
             return Observable.create(emitter -> {
                 Bitmap bitmap =
                         BitmapFactory.decodeResource(
-                                ApplicationProvider.getApplicationContext().getResources(),
+                                getApplicationContext().getResources(),
                                 CHECKER_BACKGROUND_DRAWABLE_ID);
                 emitter.onNext(bitmap);
+                FileUtils.saveBitmapToFile(bitmap, getQrCodeFilepath());
 
                 // save bitmap to test that shareQRCode generates bitmap if file not there
-                FileUtils.saveBitmapToFile(bitmap, getQrCodeFilepath());
                 emitter.onComplete();
             });
         }
 
         @Override
         public String getQrCodeFilepath() {
-            return new StoragePathProvider().getDirPath(StorageSubdirectory.SETTINGS) + File.separator + "test-collect-settings.png";
+            return getApplicationContext().getExternalFilesDir(null) + File.separator + "test-collect-settings.png";
```

Comment:
I only ran into problems with this on my emulator and then couldn't reproduce but I thought it made more sense to use a file path that had nothing to do with how the real object would work. This means it's less likely to clash with anything else and highlights that this is fake/stub.